pyanaconda/modules/localization/runtime.py

```diff
@@ -0,0 +1,178 @@
+#
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core.util import execWithRedirect
+from pyanaconda.core.constants import DEFAULT_KEYBOARD
+from pyanaconda.modules.common.errors.configuration import KeyboardConfigurationError
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.localization.localed import LocaledWrapper
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.modules.localization.installation import write_x_configuration, \
+    write_vc_configuration
+
+log = get_module_logger(__name__)
+
+
+class ConvertMissingKeyboardConfigurationTask(Task):
```

Comment:
This task is a little confusing. I wonder if the logic could be somehow automated using signals of the setters.

Reply:
After splitting missing values handling https://github.com/rhinstaller/anaconda/pull/2217/commits/69d12b4a6edd6059b410ab579a93c56a14f59841
this task becomes PopulateMissingKeyboardConfiguration and does only x_layouts <-> vc_keymap update of missing values. I think such update should be explicit to module client, especially when the task is still used as installation task (see below).